identity/validator_set.go

```diff
@@ -366,6 +425,39 @@ func (vs *ValidatorStore) GetEndBlockUpdate(ctx *ValidatorContext, req types.Req
 				}
 			}
 		}
+
+		// purge all other active validators if not among the top
+		for addr, power := range vs.lastActive {
+			addrHuman := keys.Address(addr).Humanize()
+			pub, ok := noTopValidators[addrHuman]
```

Comment:
You're just checking the NON validator map for the address to determine if it needs to be purged.